src/object-store/src/layers/prometheus.rs

```diff
@@ -425,12 +404,17 @@ impl<R> PrometheusMetricWrapper<R> {
 
 impl<R: oio::Read> oio::Read for PrometheusMetricWrapper<R> {
     fn poll_read(&mut self, cx: &mut Context<'_>, buf: &mut [u8]) -> Poll<Result<usize>> {
+        let start = Instant::now();
+
         self.inner.poll_read(cx, buf).map(|res| match res {
-            Ok(bytes) => {
+            Ok(n) => {
                 BYTES_TOTAL
                     .with_label_values(&[&self.scheme, Operation::Read.into_static()])
-                    .observe(bytes as f64);
-                Ok(bytes)
+                    .observe(n as f64);
+                REQUESTS_DURATION_SECONDS
+                    .with_label_values(&[&self.scheme, Operation::Read.into_static()])
+                    .observe(start.elapsed().as_secs_f64());
+                Ok(n)
             }
```

Comment:
It only counts the time when it succeeds (same as opendal). Should it also count the fail cases? As other functions count the time once the function is called.